### PR TITLE
Add QuickCheck coverage for 2I features.

### PR DESCRIPTION
**Pull request by seancribbs**
We didn't have decent test coverage of 2I for backends. This gives a standard suite to run against the backends that currently support it, including LevelDB and Memory (and Multi if configured properly), thanks to QuickCheck.

### CODE REVIEW

File: test/backend_eqc.erl [hunk 1]
```diff
@@ -271,17 +351,20 @@ stopped(#qcst{backend=Backend,
...
-     {history, {call, Backend, fold_keys, [fold_keys_fun(), get_fold_buffer(), g_opts(), State]}},
+     {history, {call, Backend, fold_keys, [fold_keys_fun(), get_fold_buffer(), fold_keys_opts(), State]}},
      {history, {call, Backend, fold_objects, [fold_objects_fun(), get_fold_buffer(), g_opts(), State]}},
      {history, {call, Backend, is_empty, [State]}},
      {history, {call, ?MODULE, drop, [Backend, State]}},
      {stopped, {call, Backend, stop, [State]}}
     ].
 
+precondition(_From,_To,#qcst{backend=Backend},{call, _M, fold_keys, [_FoldFun, _Acc, [{index, Bucket, _}], BeState]}) ->
+    {ok, Capabilities} = Backend:capabilities(Bucket, BeState),
```
Comment (Vagabond) [inline test/backend_eqc.erl:381]: This seems to call Backend:capabilities/2 with the symbolic state, and thus break the test.

**Comment by Vagabond** [comment]
So, this test fails for me on both multi and eleveldb backends, is this expected? Memory seems to actually pass.

**Comment by seancribbs** [comment]
@Vagabond Thanks for looking, I will debug tomorrow.

**Comment by seancribbs** [comment]
@Vagabond Another look-see? Thanks!

**Comment by Vagabond** [comment]
I haz counterexample:

https://gist.github.com/5d46a79ee2af2390c6fe

This is eleveldb without async folds enabled.

**Comment by Vagabond** [comment]
Found another with async folds on.

**Comment by Vagabond** [comment]
You need to run quite a few tests before you'll hit one, though. The default of 50 is nowhere near what is needed.

**Comment by seancribbs** [comment]
I will double-check the postconditions tomorrow, what it shrank to doesn't seem erroneous.

**Comment by seancribbs** [comment]
Yes, but I have run all backends with 250 or more and not provoked that example. Needs more time I guess.

Sean Cribbs

**Comment by Vagabond** [comment]
Perhaps your preconditions aren't strict enough to let it shrink correctly. Although that first countexample doesn't look like a bad shrink to me...

**Comment by Vagabond** [comment]
+1 to merge, now.

**Pull request merged**